SPECS/maven/maven.spec

```diff
@@ -1,65 +1,109 @@
+# Switch "sources_generation" to 1 when running a package build to generate cached sources for regular builds.
+%define sources_generation 0
+%define m2_cache_tarball_name apache-%{name}-%{version}-m2.tar.gz
+%define licenses_tarball_name apache-%{name}-%{version}-licenses.tar.gz
+
+%if ! 0%{?sources_generation}
+%define offline_build -o
+%endif
+
 Summary:        Apache Maven
 Name:           maven
-Version:        3.5.4
-Release:        13%{?dist}
+Version:        3.8.1
+Release:        1%{?dist}
 License:        ASL 2.0
-URL:            https://maven.apache.org/
-Group:          Applications/System
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
+Group:          Applications/System
+URL:            https://maven.apache.org/
 Source0:        https://archive.apache.org/dist/maven/maven-3/%{version}/source/apache-%{name}-%{version}-src.tar.gz
+# Using pre-compiled binaries because 'maven' requires itself during build-time.
 Source1:        https://archive.apache.org/dist/maven/maven-3/%{version}/binaries/apache-%{name}-%{version}-bin.tar.gz
```

Comment:
We should take a bug on this to use our OWN binaries from this build once it completes.

Reply:
That's part of a separate topic I'm planning to discuss tomorrow.

Reply:
Bug created.